extensions/ql-vscode/src/view/results/graph.tsx

```diff
@@ -1,58 +1,26 @@
 import * as React from "react";
 import { select } from "d3";
-import { ResultTableProps, jumpToLocation } from "./result-table-utils";
+import { jumpToLocation } from "./result-table-utils";
 import {
   InterpretedResultSet,
   GraphInterpretationData,
 } from "../../common/interface-types";
 import { graphviz, GraphvizOptions } from "d3-graphviz";
 import { tryGetLocationFromString } from "../../common/bqrs-utils";
-export type GraphProps = ResultTableProps & {
+import { useCallback, useEffect } from "react";
+export type GraphProps = {
   resultSet: InterpretedResultSet<GraphInterpretationData>;
+  offset: number;
+  databaseUri: string;
 };
 
 const graphClassName = "vscode-codeql__result-tables-graph";
 const graphId = "graph-results";
-export class Graph extends React.Component<GraphProps> {
-  constructor(props: GraphProps) {
-    super(props);
-  }
-
-  public render = (): JSX.Element => {
-    const { resultSet, offset } = this.props;
-    const graphData = resultSet.interpretation?.data?.dot[offset];
-
-    if (!graphData) {
-      return (
-        <>
-          <div className={graphClassName}>Graph is not available.</div>
-        </>
-      );
-    }
-
-    return (
-      <>
-        <div className={graphClassName}>
-          <strong>Warning:</strong> The Graph Viewer is not a publicly released
-          feature and will crash on large graphs.
-        </div>
-        <div id={graphId} className={graphClassName}>
-          <span>Rendering graph...</span>
-        </div>
-      </>
-    );
-  };
-
-  public componentDidMount = () => {
-    this.renderGraph();
-  };
 
-  public componentDidUpdate = () => {
-    this.renderGraph();
-  };
+export function Graph({ resultSet, offset, databaseUri }: GraphProps) {
```

Comment:
I notice that the `resultSet` and `offset` props are only used to calculate `graphData`. I think we could simplify the props by passing in `graphData` directly. It would look something like https://github.com/github/vscode-codeql/commit/fea93c2faf3c0c71c06f051e62c3c40528c5ffdb

Reply:
Very good idea! Looks a lot better now! (and thanks for the branch!)